internal/profile/profile_test.go

```diff
@@ -0,0 +1,79 @@
+package profile_test
+
+import (
+	"testing"
+
+	gh_profile "github.com/pkg/profile"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/cyberark/secretless-broker/internal/profile"
+)
+
+type stopper struct {
+	eventChan chan struct{}
+}
+
+func (s *stopper) Stop() {
+	s.eventChan <- struct{}{}
+}
+
+func TestProfile(t *testing.T) {
+	t.Run("New profile only allows cpu and memory types", func(t *testing.T) {
+		for _, profileType := range []string{"cpu", "memory"} {
+			profile, err := profile.New(profileType)
+
+			assert.NotNil(t, profile)
+			assert.Nil(t, err)
+		}
+
+		for _, profileType := range []string{"foo", "bar"} {
+			profile, err := profile.New(profileType)
+
+			assert.Nil(t, profile)
+
+			// Sanity check
+			assert.NotNil(t, err)
+			if err == nil {
+				t.Fail()
+			}
+
+			assert.Contains(t, err.Error(), "Invalid profile type")
+			assert.Contains(t, err.Error(), profileType)
+		}
+	})
+
+	t.Run("New profile delegates to profile imterface object", func(t *testing.T) {
+		startEventChan := make(chan struct{}, 10)
+		stopEventChan := make(chan struct{}, 10)
+
+		mockProfiler := func(profiles ...func(*gh_profile.Profile)) interface{ Stop() } {
+			startEventChan <- struct{}{}
+
+			return &stopper{
+				eventChan: stopEventChan,
+			}
+		}
```

Comment:
There's a (imo slightly cleaner) way to do this that avoids the `stopper` struct (which is A. not needed and B. a little misleading in that the `-er` naming is typically used for interfaces, whereas this is an impl struct).

In any case, we already have 
```go
type StartProfilingFunc func(...func(*profile.Profile)) interface{ Stop() }
```
and we can complete the symmetry here with:
```go
type StopProfilingFunc func()
```
and then use the "function type calling itself" trick to make it conform to `interface{ Stop() }`:
```go
func (stop StopProfilingFunc) Stop() {
    stop()
}
```
Now we may elimnate the current `stopper` struct, and instead rely on a closure to get hold of our `stopEventChan`, and replace lines 52-54 with:
```go
stop := func() {
    stopEventChan <- struct{}{}
}
return StopProfilingFunc(stop)
```
POC of the concept: https://play.golang.org/p/Zfw3bJvnUP_Q